syncer/syncer_test.go

```diff
@@ -239,20 +224,6 @@ func TestSynchronize_FetchLayerDataFailed(t *testing.T) {
 	require.True(t, ts.syncer.ListenToATXGossip())
 	require.False(t, ts.syncer.ListenToGossip())
 	require.False(t, ts.syncer.IsSynced(context.TODO()))
-
-	syncedCh := ts.syncer.RegisterForSynced(context.TODO())
-	select {
-	case <-syncedCh:
-		require.Fail(t, "node should not be synced")
-	case <-time.After(1 * time.Millisecond):
-	}
-
-	atxSyncedCh := ts.syncer.RegisterForATXSynced()
-	select {
-	case <-atxSyncedCh:
-	case <-time.After(1 * time.Second):
-		require.Fail(t, "node should be atx synced")
-	}
```

Comment:
I think that this assert is also required; If `ListenToATXGossip` returns `true` `RegisterForATXSynced` should return a closed channel.

Reply:
RegisterForATXSynced is just a method. you already tested it in `TestSyncer_setATXSyncedTwice_NoError` that it behaves correctly. `TestFromSyncedToNotSynced` also tested that it channel get notified when the node is synced. it does not need to be checked in EVERY test.

Reply:
added async tests in TestSynchronize_AllGood